packages/kokkos-kernels/src/sparse/impl/KokkosSparse_cluster_gauss_seidel_impl.hpp

```diff
@@ -529,7 +529,7 @@ namespace KokkosSparse{
       struct BuildCrossClusterMaskFunctor
       {
         BuildCrossClusterMaskFunctor(Rowmap& rowmap_, Colinds& colinds_, nnz_view_t& clusterOffsets_, nnz_view_t& clusterVerts_, nnz_view_t& vertClusters_, bitset_t& mask_)
-          : rowmap(rowmap_), colinds(colinds_), clusterOffsets(clusterOffsets_), clusterVerts(clusterVerts_), vertClusters(vertClusters_), mask(mask_)
+          : numRows(rowmap_.extent(0) - 1), rowmap(rowmap_), colinds(colinds_), clusterOffsets(clusterOffsets_), clusterVerts(clusterVerts_), vertClusters(vertClusters_), mask(mask_)
```

Comment:
If the matrix has zero rows, `numRows` could be negative.  That doesn't break the code below, though.

Reply:
@mhoemmen Yes, every functor is a loop over rows (which is passed in, not determined from rowmap length) so they should be fine. In KokkosKernels I added a test that runs MTSGS and cluster MTSGS on zero-row matrices, with both 0-length and 1-length rowmaps. That ran clean in valgrind.